### ISSUE TITLE
Proof environments are properly rendered in PDF as well

### ISSUE THREAD

**Issue by andry-tino** [executablebooks/mystmd#588]
Rendering to PDF an MD document which has a `{prf:theorem}` environment in it results in an error and a PDF generated which does not contain the theorem. Nonetheless, the HTML rendering works fine.

Using latest: 1.1.15

## Proposal
Create a basic MD file which contains a proof:

```
---
title: My file
authors:
    - name: Me
      email: me@me.me
keywords: [aaa, bbb]
date: 2023-09-10
exports:
  - format: pdf
    template: arxiv_nips
    output: exports/myfile.pdf
---

+++ { "part": "abstract" }
A simple test
+++

Let us consider an theorem,

:::{prf:theorem} A theorem
This is an example of how to hide the content of a directive.
:::
```

Export to PDF:

    myst build myfile.md --pdf

Get this error:

> ⛔️ Unhandled LaTeX conversion for node of "proof"

## Proposed solution
No such error should be shown and proofs and theorems should be available in PDF as well.

## Additional notes
If trying to export to html:

    myst build --html

All good.

**Comment by fwkoch** [comment]
Just tagging that this issue is the same pattern as this one: https://github.com/executablebooks/mystmd/issues/584 (I know you are probably aware of this since you made that issue too @andry-tino 😀 Thank you!)

**Comment by andry-tino** [comment]
Hi @fwkoch thanks for your message. Mmm I would not qualify them as same issue. I am already working on #584 :)

This issue is about a different feature. They both target different syntax features to be properly supported. That is why I believe they are not exactly the same. I can see from code that there are other unsupported features concerning the TEX rendering. Let us address them one by one :)

**Comment by fwkoch** [comment]
Yes, totally! Different features, just similar "unsupported tex nodes" pattern. One by one is perfect. 😀

**Comment by andry-tino** [comment]
@fwkoch As I complete the fix on glossary, then move to proof envs, I will try discovering other nodes not supported and create issues for them. Seems like a good way to get acquainted with the codebase :)

**Comment by andry-tino** [comment]
@fwkoch can I also please get this one assigned to me?

**Comment by andry-tino** [comment]
@fwkoch The PR is in but I would love you to give it a try and close this issue after you verify it actually works :)

**Comment by fwkoch** [comment]
Working nicely for me!

![image](https://github.com/executablebooks/mystmd/assets/9453731/4442b908-2163-48c8-b3a7-af7e0c81f488)

I made one more tiny PR - `{prf:corollary}` and `{proof}` were skipped over in the `proofHandler` (the former was just missing from the `switch` statement, the latter left `kind` undefined): #675

**Comment by fwkoch** [comment]
That last little pr is landed, and now this is in the latest myst release! Thanks again @andry-tino
